### ISSUE TITLE
Can't copy a single file to a sub folder in Azure Files

### ISSUE THREAD

**Issue by maroony** [Azure/azure-storage-azcopy#410]
### Which version of the AzCopy was used? 
10.1.2

### Which platform are you using? (ex: Windows, Mac, Linux)
Linux, WSL

### What command did you run?
`azcopy cp file.txt "${fileStorageUrl}${storage}/folder/subfolder/?${sasToken}"`

### What problem was encountered?
log output: `RESPONSE Status: 400 The specifed resource name contains invalid characters.`

### How can we reproduce the problem in the simplest way?
see the command above

### Have you found a mitigation/solution?
no

**Comment by JohnRusk** [comment]
Possibly related to (but not identical to) #397

**Comment by JohnRusk** [comment]
@maroony Thanks for logging this.  One possible workaround you might try (maybe a long shot) but worth trying, is to try the same command but without the "/" at the end of "subfolder". I.e. like this:

```
azcopy cp file.txt "${fileStorageUrl}${storage}/folder/subfolder?${sasToken}"
```
I don't know yet exactly where the bug is, so I don't know if that may avoid it or not, but its worth trying if you have time.

**Comment by maroony** [comment]
@JohnRusk If you remove the last "/", the `file.txt` will copied to the folder `folder` *and* renamed to `subfolder`. So a possible workarround would be

```
azcopy cp file.txt "${fileStorageUrl}${storage}/folder/subfolder/file.txt?${sasToken}"
```

In my scenario I can't you wildcards.

**Comment by JohnRusk** [comment]
Good point. Thanks!

**Comment by adreed-msft** [comment]
Hi there, this was patched in #451 and is now present in AzCopy v10.2.x. Enjoy!
